src/main/java/teammates/sqllogic/core/FeedbackResponsesLogic.java

```diff
@@ -183,5 +189,195 @@ public boolean areThereResponsesForQuestion(UUID questionId) {
      */
     public boolean hasResponsesForCourse(String courseId) {
         return frDb.hasResponsesForCourse(courseId);
+
+    }
+
+    /**
+     * Deletes all feedback responses involved an entity, cascade its associated comments.
+     */
+    public void deleteFeedbackResponsesForCourseCascade(String courseId, String entityEmail) {
+        // delete responses from the entity
+        List<FeedbackResponse> responsesFromStudent =
+                getFeedbackResponsesFromGiverForCourse(courseId, entityEmail);
+        for (FeedbackResponse response : responsesFromStudent) {
+            frDb.deleteFeedbackResponse(response);
```

Comment:
Without looking at the entity annotations, it can be easy to mistake this code to have not cascaded deletion to feedback response comments. Perhaps we can elaborate in the comments?

Reply:
Jingjing, is this good?

```java
// Delete responses from the entity along with its associated comments. (here!)
List<FeedbackResponse> responsesFromStudent =
        getFeedbackResponsesFromGiverForCourse(courseId, entityEmail);
for (FeedbackResponse response : responsesFromStudent) {
    frDb.deleteFeedbackResponse(response);
}

// Delete responses to the entity along with its associated comments. (here!)
List<FeedbackResponse> responsesToStudent =
        getFeedbackResponsesForRecipientForCourse(courseId, entityEmail);
for (FeedbackResponse response : responsesToStudent) {
    frDb.deleteFeedbackResponse(response);
}
```

Reply:
Sorry for the confusion! I was referring to how the cascading of delete is handled by Hibernate. So maybe something along the lines of "Deletion will automatically be cascaded to the comments, handled by Hibernate using the OnDelete annotation"?